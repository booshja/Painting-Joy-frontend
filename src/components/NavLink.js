// dependencies
import React from "react";
import styled from "styled-components";
// components
import { NavLink as ReactNavLink } from "react-router-dom";

const StyledLink = styled(ReactNavLink)`
    font-family: "News Cycle", sans-serif;
    font-weight: 400;
    font-size: 1.4rem;
    letter-spacing: 2px;

    &:hover:not(.active) {
        transform: translateY(-1px);
    }

    &.active {
        font-weight: 700;
        color: #207070;
    }
`;

const NavLink = ({ to, name }) => {
    return (
        <StyledLink exact={to === "/" ? true : false} to={`${to}`}>
            {name}
        </StyledLink>
    );
};

export default NavLink;
