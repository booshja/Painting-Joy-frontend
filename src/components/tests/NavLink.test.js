import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NavLink from "../NavLink";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <NavLink to="/testlink" name="Test Link" />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <NavLink to="/testlink" name="Test Link" />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText } = render(
        <BrowserRouter>
            <NavLink to="/testlink" name="Test Link" />
        </BrowserRouter>
    );

    getByText("Test Link");
});
