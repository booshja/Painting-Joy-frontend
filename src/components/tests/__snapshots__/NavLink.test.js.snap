// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="sc-bdnxRM jTcfkt"
    href="/testlink"
  >
    Test Link
  </a>
</DocumentFragment>
`;
